src/Block/FileBlock.php

```diff
@@ -19,8 +21,51 @@ class FileBlock extends BaseElement
 
     private static $plural_name = 'files';
 
+    private static $icon = 'font-icon-block-file';
+
     public function getType()
     {
         return _t(__CLASS__ . '.BlockType', 'File');
     }
+
+    public function getSummary()
+    {
+        if ($this->File() && $this->File()->exists()) {
+            return $this->getSummaryThumbnail() . $this->File()->Title;
+        }
+        return '';
+    }
+
+    /**
+     * Return a thumbnail of the file, if it's an image. Used in GridField preview summaries.
+     *
+     * @return string|null
+     */
+    public function getSummaryThumbnail()
+    {
+        if ($this->File() && $this->File()->exists()) {
+            $template = '<span class="elemental-preview__thumbnail-image">%s</span>';
```

Comment:
HTML in a PHP file? yikes - have you thought about moving this to an Include and rendering that instead?

Reply:
That sounds beautiful